perses/distributed/feptasks.py

```diff
@@ -225,7 +229,7 @@ def run_protocol(thermodynamic_state, sampler_state, ne_mc_move, topology, n_ite
         ne_mc_move.apply(thermodynamic_state, sampler_state)
 
         #record the positions as a result
-        trajectory_positions[iteration, :, :] = sampler_state.positions
+        trajectory_positions[iteration, :, :] = sampler_state.positions[atom_indices, :]
```

Comment:
Isn't `sampler_state.positions` a unit-bearing quantity that you are forcing into a numpy array (`trajectory_positions`) without explicit unit conversion? This will break if `SamplerState` default units ever change. Can you at least explicitly convert all these quantities into the appropriate MDTraj units explicitly?

Reply:
Actually, it may be even better to add a method to `SamplerState` like `.to_mdtraj_trajectory(topology, atom_indices=None)` where it will automatically populate a `Topology` object for you using the specified atom indices.